apps/studio/tests/unit/lib/editor.spec.js

```diff
@@ -0,0 +1,130 @@
+import {
+  makeDBHint,
+  parseDBHintTable,
+  parseDBHintTables,
+  queryTable,
+  findTablesBySchema,
+  pushTablesToResult,
+  sanitizeTableName,
+} from "../../../src/lib/editor";
+import { tableOrViews } from "../../fixtures/tables";
+
+describe("lib/editor", () => {
```

Comment:
Please correct me if I'm wrong here, but it looks like this won't work if the DB has schemas, and no schema is provided in the query?

Postgres by default uses the `public` schema, and SQL Server uses `dbo`, and not specifying this is totally legit, and really common honestly.

A good unit test would be to check that it can resolve:

- `schema.table`
- `schema.table.column`
- `table`
- `table.column`

The complex bit is that `a.b` could be either `schema.table` or `table.column`.

It gets a little more complex in that there could be both `schema: a`, and `table: public.a`, so the hint system would need to provide some ability to handle this situation. So if someone types `a.`, it should probably autocomplete both table names from `schema: a` and column names from `table: public.a`.

Default schema is [also provided in the store as a getter](https://github.com/beekeeper-studio/beekeeper-studio/blob/e405db30297950a4295c9d1a15e78aa2713e9a45/apps/studio/src/store/index.ts#L104), so should be easy to integrate if not already?

Reply:
Oh, you've got a point. I forgot to add the default schema.

I also think this part is a little bit confusing because there are [two ways](https://github.com/beekeeper-studio/beekeeper-studio/pull/1610/files#diff-8ada96660a46ee9c6784d974ac1043ced5db7c7f1be4eab25c9e25ecb43fbd3bR66-R70) to find tables right now: one by looking up the key in `tables` object, the other one is by using the `hintTables` array. I add the second one because the `tables` can overwrite the same table names that have different schemas, while the `hintTables` can not.

I'll try to add the test to clarify this one and fix the other feedbacks soon.

Reply:
Your test suggestions should now be covered by the [queryTable test](https://github.com/beekeeper-studio/beekeeper-studio/pull/1610/files#diff-4e044bf2110531d177d19691735abb6cea1e27c92e792eed2734814d8b274ab9R144). As long as it resolves the `schema.table`, it should be able to resolve `schema.table.column` too which depends on the column lookup [here](https://github.com/beekeeper-studio/beekeeper-studio/pull/1610/files#diff-8bd68c134db51e45491610ee0f614c52db0be8412bf23eded25bf6aeffa086a9R918). It should also work when schema is not provided or `undefined` [(test)](https://github.com/beekeeper-studio/beekeeper-studio/pull/1610/files#diff-4e044bf2110531d177d19691735abb6cea1e27c92e792eed2734814d8b274ab9R126-R142).